lib/install/management/configure.go

```diff
@@ -431,16 +433,15 @@ func extractSecretFromFile(rc io.ReadCloser) (string, error) {
 	return "", errSecretKeyNotFound
 }
 
-// GuestInfoSecret downloads the VCH's .vmx file and returns the GuestInfoSecretKey
-// value. This function expects the datastore in the dispatcher's session to be set.
-func (d *Dispatcher) GuestInfoSecret(vchName string) (*extraconfig.SecretKey, error) {
+// GuestInfoSecret downloads the VCH's .vmx file and returns the GuestInfoSecretKey value.
+func (d *Dispatcher) GuestInfoSecret(vchName, vmPath string, ds *object.Datastore) (*extraconfig.SecretKey, error) {
 	defer trace.End(trace.Begin(""))
 
-	if d.session.Datastore == nil {
+	if ds == nil {
```

Comment:
hmm.. I prefer to keep an error here, to make sure no matter how user will get an error message, instead of a panic